radixdlt-java/radixdlt-java/src/main/java/com/radixdlt/client/lib/api/token/Amount.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (C) Copyright 2021 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ */
+
+package com.radixdlt.client.lib.api.token;
+
+import com.radixdlt.utils.UInt256;
+
+public interface Amount {
```

Comment:
I think there's a lot of issues with this interfaces. Representing all forms of subunits in the same `UInt256` I think makes things confusing. Also there's no overflow or underflow protection. Would rather use something more similar to what is currently `com.radixdlt.application.tokens.Amount.java`

Reply:
The idea behind this interface is to make code readable closer to plain English: `amount(10).tokens()` or `amount(100).nanos()`. It is also closer to how we usually express the amounts in real life - "10 bucks" or "15 cents". 
Since this interface is part of the client library, I feel that it will be more convenient for end-users.

Since there is only a constructor from `long`, I doubt that overflow/underflow is an issue.